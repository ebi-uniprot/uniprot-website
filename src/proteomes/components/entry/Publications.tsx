import { FC } from 'react';
import { DataList } from 'franklin-sites';

import CitationCard from '../../../supporting-data/citations/components/results/CitationsCard';
import TaxonomyView from '../../../shared/components/entry/TaxonomyView';

import { getCitationItemId } from '../../../supporting-data/citations/utils';

import { Citation } from '../../../supporting-data/citations/adapters/citationsConverter';
import { ProteomesAPIModel } from '../../adapters/proteomesConverter';

const Publications: FC<Pick<ProteomesAPIModel, 'citations' | 'taxonomy'>> = ({
  citations,
  taxonomy,
}) => {
  if (!citations?.length) {
    return null;
  }

  return (
    <section>
      <h2>
        {'Publications for '}
        <TaxonomyView data={taxonomy} noLink />
      </h2>
      <DataList
        getIdKey={getCitationItemId}
        data={citations}
        dataRenderer={(citation: Citation) => (
          <CitationCard data={{ citation }} />
        )}
      />
    </section>
  );
};

export default Publications;
