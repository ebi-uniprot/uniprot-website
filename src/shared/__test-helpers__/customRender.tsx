/**
 * This tries to follow what is detailed here:
 * https://testing-library.com/docs/react-testing-library/setup#custom-render
 */
import { render, RenderOptions } from '@testing-library/react';
import { Component, Dispatch, ReactElement, ReactNode } from 'react';
import { HelmetProvider } from 'react-helmet-async';
import {
  createMemoryHistory,
  LocationState,
  MemoryHistory,
  Route,
  Router,
} from 'react-router';
import { JsonValue, SetRequired } from 'type-fest';

import { MessagesState } from '../../messages/state/messagesInitialState';
import { MessagesAction } from '../../messages/state/messagesReducers';
import databaseInfo from '../../uniprotkb/utils/__tests__/__mocks__/databaseInfo';
import { getDatabaseInfoMaps } from '../../uniprotkb/utils/database';
import {
  MessagesDispatchContext,
  MessagesStateContext,
} from '../contexts/Messages';
import {
  DatabaseInfoMapsContext,
  UniProtDataVersionContext,
} from '../contexts/UniProtData';
import { JobAction } from '../workers/jobs/state/actionHandler';
import { JobsState } from '../workers/jobs/state/jobsInitialState';

type ExtraRenderOptions = {
  // For react-router
  history?: MemoryHistory<LocationState>;
  path?: string;
  // For custom user preferences (used by useLocalStorage)
  initialLocalStorage?: Record<string, JsonValue>;
  // For app context
  messagesState?: MessagesState;
  messagesDispatch?: jest.Mock<Dispatch<MessagesAction>>;
  JobsState?: JobsState;
  toolsDispatch?: jest.Mock<Dispatch<JobAction>>;
};

type WrapperProps = { children: ReactNode } & RenderOptions &
  SetRequired<
    ExtraRenderOptions,
    | 'history'
    | 'initialLocalStorage'
    | 'messagesState'
    | 'messagesDispatch'
    | 'JobsState'
    | 'toolsDispatch'
  >;

const dbInfoMaps = getDatabaseInfoMaps(databaseInfo);
const uniProtDataVersion = {
  releaseNumber: '2023_01',
  releaseDate: new Date('01-March-2023'),
};

class Wrapper extends Component<WrapperProps> {
  constructor(props: WrapperProps) {
    super(props);
    for (const [key, value] of Object.entries(props.initialLocalStorage)) {
      window.localStorage.setItem(key, JSON.stringify(value));
    }
  }

  // eslint-disable-next-line class-methods-use-this
  componentWillUnmount() {
    window.localStorage.clear();
  }

  render() {
    const { children, path, history, messagesDispatch, messagesState } =
      this.props;
    return (
      <HelmetProvider>
        <Router history={history}>
          <MessagesDispatchContext.Provider value={messagesDispatch}>
            <MessagesStateContext.Provider value={messagesState}>
              <DatabaseInfoMapsContext.Provider value={dbInfoMaps}>
                <UniProtDataVersionContext.Provider value={uniProtDataVersion}>
                  {path ? (
                    <Route path={path} render={() => children} />
                  ) : (
                    children
                  )}
                </UniProtDataVersionContext.Provider>
              </DatabaseInfoMapsContext.Provider>
            </MessagesStateContext.Provider>
          </MessagesDispatchContext.Provider>
        </Router>
      </HelmetProvider>
    );
  }
}

const customRender = (
  ui: ReactElement,
  {
    route = '',
    path,
    history = createMemoryHistory({ initialEntries: [route] }),
    initialLocalStorage = {},
    messagesState = {},
    messagesDispatch = jest.fn(),
    JobsState = {},
    toolsDispatch = jest.fn(),
    ...options
  }: RenderOptions &
    ExtraRenderOptions & {
      route?: string;
    } = {}
) => ({
  ...render(ui, {
    wrapper: (props) => (
      <Wrapper
        path={path}
        history={history}
        initialLocalStorage={initialLocalStorage}
        messagesState={messagesState}
        messagesDispatch={messagesDispatch}
        JobsState={JobsState}
        toolsDispatch={toolsDispatch}
        {...props}
      />
    ),
    ...options,
  }),
  history,
  messagesDispatch,
  toolsDispatch,
});

export default customRender;
