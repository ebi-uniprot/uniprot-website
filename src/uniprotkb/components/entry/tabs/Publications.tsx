import { useState, useEffect, useMemo, Fragment } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { Card, Loader, DataListWithLoader, InfoList } from 'franklin-sites';
import { Except, SetRequired, Simplify } from 'type-fest';
import { groupBy, capitalize } from 'lodash-es';
import { InfoListItem } from 'franklin-sites/dist/types/components/info-list';

import ExternalLink from '../../../../shared/components/ExternalLink';
import ErrorHandler from '../../../../shared/components/error-pages/ErrorHandler';

import useDataApi from '../../../../shared/hooks/useDataApi';
// import usePrefetch from '../../../shared/hooks/usePrefetch';
import useDatabaseInfoMaps from '../../../../shared/hooks/useDatabaseInfoMaps';

import EntryTypeIcon from '../../../../shared/components/entry/EntryTypeIcon';
import LiteratureCitation from '../../../../supporting-data/citations/components/LiteratureCitation';

import { addBlastLinksToFreeText } from '../../../../shared/utils/utils';
import getNextURLFromHeaders from '../../../../shared/utils/getNextURLFromHeaders';
import { getIdKeyForNamespace } from '../../../../shared/utils/getIdKey';
import { getParamsFromURL } from '../../../utils/resultsUtils';
import { processUrlTemplate } from '../../../../shared/utils/xrefs';

import { Location, LocationToPath } from '../../../../app/config/urls';

import {
  CitationsAPIModel,
  Reference,
} from '../../../../supporting-data/citations/adapters/citationsConverter';
import { Namespace } from '../../../../shared/types/namespaces';
import { SearchResults } from '../../../../shared/types/results';
import apiUrls from '../../../config/apiUrls/apiUrls';

type PublicationsReferenceProps = {
  references: Reference[];
  accession: string;
};

const PublicationReference = ({
  references,
  accession,
}: PublicationsReferenceProps) => {
  const databaseInfoMaps = useDatabaseInfoMaps();

  const infoListWithContent = references.map((reference) => {
    const {
      referencePositions,
      referenceComments,
      source,
      sourceCategories,
      communityAnnotation,
      annotation,
    } = reference;

    const databaseInfo =
      source && databaseInfoMaps?.databaseToDatabaseInfo[source.name];
    let url =
      source?.id &&
      processUrlTemplate(databaseInfo?.uriLink, { id: source.id });
    if (source?.name === 'GeneRif') {
      url = `https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=${source.id}`;
    }
    // if (source?.name === 'GAD') => don't inject a link
    // This DB is dead so we only display the accession, but no link to point to

    const groupedReferenceComments = groupBy(referenceComments, 'type');

    const infoListData = [
      {
        title: 'Cited for',
        content:
          referencePositions &&
          addBlastLinksToFreeText(referencePositions, accession).map(
            (item, i) => (
              // eslint-disable-next-line react/no-array-index-key
              <Fragment key={i}>
                {i > 0 && ', '}
                {item}
              </Fragment>
            )
          ),
      },
      ...Object.entries(groupedReferenceComments).map(([type, comments]) => {
        // Capitalise title
        const title = capitalize(type);
        return {
          title,
          content: (
            <>
              {comments.map((comment, i) => (
                // eslint-disable-next-line react/no-array-index-key
                <Fragment key={i}>
                  {i > 0 && ', '}
                  {type === 'STRAIN' ? (
                    <Link
                      to={{
                        pathname: LocationToPath[Location.UniProtKBResults],
                        search: `query=strain:"${comment.value}"`,
                      }}
                    >
                      {comment.value}
                    </Link>
                  ) : (
                    comment.value
                  )}
                </Fragment>
              ))}
            </>
          ),
        };
      }),
      {
        title: 'Annotation',
        // both mutually exclusive
        content: annotation || communityAnnotation?.comment,
      },
      {
        title: 'Function',
        content: communityAnnotation?.function,
      },
      {
        title: 'Disease',
        content: communityAnnotation?.disease,
      },
      {
        title: 'Categories',
        content: sourceCategories?.join(', '),
      },
      {
        title: 'Source',
        content: source && (
          <>
            <EntryTypeIcon entryType={source.name} />
            {url ? (
              <>
                <EntryTypeIcon entryType="computationally mapped" />
                {source.name}:<ExternalLink url={url}>{source.id}</ExternalLink>
              </>
            ) : (
              <span>
                {source.name}
                {source.name === 'ORCID' && source.id ? (
                  <>
                    {': '}
                    <ExternalLink url={`https://orcid.org/${source.id}`}>
                      {source.id}
                    </ExternalLink>
                    {' ('}
                    <ExternalLink
                      url={`//community.uniprot.org/bbsub/bbsubinfo.html?accession=${accession}`}
                    >
                      see community submission
                    </ExternalLink>
                    ).
                  </>
                ) : (
                  <>{source.id && `:${source.id}`}</>
                )}
              </span>
            )}
          </>
        ),
      },
    ];
    return infoListData;
  });

  // Merging all of them into one
  let mergedInfoList: InfoListItem[] = [];
  infoListWithContent.forEach((arr) => {
    if (mergedInfoList.length) {
      arr.forEach((obj, i) => {
        if (
          obj.title === mergedInfoList[i].title &&
          obj.content !== undefined &&
          mergedInfoList[i].content !== obj.content
        ) {
          mergedInfoList[i].content = (
            <>
              {mergedInfoList[i].content}
              {mergedInfoList[i].content && ', '}
              {obj.content}
            </>
          );
        }
      });
    } else {
      mergedInfoList = [...arr];
    }
  });

  return (
    <InfoList infoData={mergedInfoList} isCompact className="text-block" />
  );
};

const getIdKey = getIdKeyForNamespace(Namespace.citations);

const cardRendererFor =
  (accession: string) =>
  (
    data: Simplify<
      Except<CitationsAPIModel, 'references'> &
        Required<Pick<CitationsAPIModel, 'references'>>
    >
  ) =>
    (
      <Card>
        <LiteratureCitation data={data} headingLevel="h3" linkToEntry>
          <PublicationReference
            references={data.references}
            accession={accession}
          />
        </LiteratureCitation>
      </Card>
    );

const hasReference = (
  data: CitationsAPIModel
): data is SetRequired<CitationsAPIModel, 'references'> =>
  Boolean(data.references?.length);

type PublicationsProps = { accession: string };

const Publications = ({ accession }: PublicationsProps) => {
  const { search } = useLocation();
  const [{ selectedFacets }] = getParamsFromURL(search);
  const initialUrl = apiUrls.publications.entryPublications({
    accession,
    selectedFacets,
  });

  const [url, setUrl] = useState(initialUrl);
  const [allResults, setAllResults] = useState<CitationsAPIModel[]>([]);
  const [metaData, setMetaData] = useState<{
    total: number;
    nextUrl?: string;
  }>(() => ({ total: 0, nextUrl: undefined }));
  // usePrefetch(metaData.nextUrl);

  const { data, loading, status, error, headers } =
    useDataApi<SearchResults<CitationsAPIModel>>(url);

  const resultsWithReferences = useMemo(
    () => allResults.filter(hasReference),
    [allResults]
  );

  useEffect(() => {
    setAllResults([]);
    setMetaData({ total: 0, nextUrl: undefined });
    setUrl(initialUrl);
  }, [initialUrl]);

  useEffect(() => {
    if (!data) {
      return;
    }
    const { results } = data;
    setAllResults((allRes) => [...allRes, ...results]);
    setMetaData(() => ({
      total: +(headers?.['x-total-results'] || 0),
      nextUrl: getNextURLFromHeaders(headers),
    }));
  }, [data, headers]);

  const cardRenderer = useMemo(() => cardRendererFor(accession), [accession]);

  if (error) {
    return <ErrorHandler status={status} />;
  }

  if (allResults.length === 0 && loading) {
    return <Loader />;
  }

  const { total, nextUrl } = metaData;

  return (
    <section>
      <h2>Publications for {accession}</h2>
      <DataListWithLoader
        getIdKey={getIdKey}
        data={resultsWithReferences}
        dataRenderer={cardRenderer}
        onLoadMoreItems={() => nextUrl && setUrl(nextUrl)}
        loaderComponent={<Loader />}
        hasMoreData={total > allResults.length}
      />
    </section>
  );
};

export default Publications;
