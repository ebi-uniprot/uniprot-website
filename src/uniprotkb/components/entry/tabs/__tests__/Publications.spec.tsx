import { screen } from '@testing-library/react';

import customRender from '../../../../../shared/__test-helpers__/customRender';
import entryPublicationData from '../../__tests__/__mocks__/entryPublicationsData';
import { PublicationReference, PublicationSource } from '../Publications';

describe('PublicationReference', () => {
  it('should render correctly', () => {
    const { asFragment } = customRender(
      <PublicationReference
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        references={entryPublicationData.results[0].references!}
        accession="O43865"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('PublicationSource', () => {
  describe('UniProtKB', () => {
    it('renders UniProtKB reviewed', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'UniProtKB reviewed (Swiss-Prot)' }}
        />
      );
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders UniProtKB unreviewed', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'UniProtKB unreviewed (TrEMBL)' }}
        />
      );
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Community annotations', () => {
    it('renders ORCID links', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'ORCID', id: '0009-0004-6150-6467' }}
        />
      );
      expect(screen.getAllByRole('link')).toHaveLength(2);
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders ORCID links', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'ORCID', id: 'Anonymous' }}
        />
      );
      expect(screen.getAllByRole('link')).toHaveLength(1);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Normal xref', () => {
    it('renders correctly for normal xrefs', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'IntAct', id: 'P12345' }}
        />
      );
      const link = screen.getByRole('link');
      expect(link).toHaveAttribute('href', expect.stringContaining('P12345'));
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders but without links for unknown DBs', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'not-a-db', id: 'ID1234' }}
        />
      );
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Exceptions', () => {
    it('renders correctly for GeneRif', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'GeneRif', id: '351' }}
        />
      );
      expect(screen.getByRole('link')).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly for MGI', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'MGI', id: '1860086' }}
        />
      );
      const link = screen.getByRole('link');
      expect(link).toHaveAttribute('href', expect.stringContaining('MGI:'));
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly for IEDB', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'IEDB', id: 'P05067' }}
        />
      );
      expect(screen.getByRole('link')).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly for IMPC', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'IMPC', id: 'MGI:1860086' }}
        />
      );
      expect(screen.getByRole('link')).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly for GAD', () => {
      const { asFragment } = customRender(
        <PublicationSource
          accession="P12345"
          source={{ name: 'GAD', id: '117462' }}
        />
      );
      expect(asFragment()).toMatchSnapshot();
      expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });
  });
});
