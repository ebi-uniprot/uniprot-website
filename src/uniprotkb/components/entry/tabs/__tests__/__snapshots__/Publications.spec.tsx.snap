// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationReference should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="text-block info-list"
  >
    <li>
      <div
        class="decorated-list-item decorated-list-item--compact"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Cited for
        </div>
        <div
          class="decorated-list-item__content"
        >
          NUCLEOTIDE SEQUENCE [MRNA] (ISOFORM 1), TISSUE SPECIFICITY, DEVELOPMENTAL STAGE
        </div>
      </div>
    </li>
    <li>
      <div
        class="decorated-list-item decorated-list-item--compact"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Categories
        </div>
        <div
          class="decorated-list-item__content"
        >
          Sequences, Expression
        </div>
      </div>
    </li>
    <li>
      <div
        class="decorated-list-item decorated-list-item--compact"
      >
        <div
          class="decorated-list-item__title tiny"
        >
          Source
        </div>
        <div
          class="decorated-list-item__content"
        >
          <span
            class="entry-title__status icon--reviewed"
            title="This marks a reviewed UniProtKB entry"
          >
            <test-file-stub />
          </span>
          UniProtKB reviewed (Swiss-Prot)
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`PublicationSource Community annotations renders ORCID links 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--community-annotation"
    title="Community annotation"
  >
    <test-file-stub />
  </span>
  ORCID: 
  <a
    class="external-link"
    href="https://orcid.org/0009-0004-6150-6467"
    rel="noopener"
    target="_blank"
  >
    0009-0004-6150-6467
    <test-file-stub
      data-testid="external-link-icon"
      width="12.5"
    />
  </a>
   (
  <a
    class="external-link"
    href="//community.uniprot.org/bbsub/bbsubinfo.html?accession=P12345"
    rel="noopener"
    target="_blank"
  >
    see community submission
    <test-file-stub
      data-testid="external-link-icon"
      width="12.5"
    />
  </a>
  ).
</DocumentFragment>
`;

exports[`PublicationSource Community annotations renders ORCID links 2`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--community-annotation"
    title="Community annotation"
  >
    <test-file-stub />
  </span>
  ORCID: Anonymous (
  <a
    class="external-link"
    href="//community.uniprot.org/bbsub/bbsubinfo.html?accession=P12345"
    rel="noopener"
    target="_blank"
  >
    see community submission
    <test-file-stub
      data-testid="external-link-icon"
      width="12.5"
    />
  </a>
  ).
</DocumentFragment>
`;

exports[`PublicationSource Exceptions renders correctly for GAD 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--computational-annotation"
    title="Computational annotation"
  >
    <test-file-stub />
  </span>
  GAD: 117462
</DocumentFragment>
`;

exports[`PublicationSource Exceptions renders correctly for GeneRif 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--computational-annotation"
    title="Computational annotation"
  >
    <test-file-stub />
  </span>
  GeneRif: 
  <a
    class="external-link"
    href="https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=351"
    rel="noopener"
    target="_blank"
  >
    351
    <test-file-stub
      data-testid="external-link-icon"
      width="12.5"
    />
  </a>
</DocumentFragment>
`;

exports[`PublicationSource Exceptions renders correctly for IEDB 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--computational-annotation"
    title="Computational annotation"
  >
    <test-file-stub />
  </span>
  IEDB: 
  <a
    class="external-link"
    href="https://iedb.org/antigen/UNIPROT:P05067"
    rel="noopener"
    target="_blank"
  >
    P05067
    <test-file-stub
      data-testid="external-link-icon"
      width="12.5"
    />
  </a>
</DocumentFragment>
`;

exports[`PublicationSource Exceptions renders correctly for IMPC 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--computational-annotation"
    title="Computational annotation"
  >
    <test-file-stub />
  </span>
  IMPC: 
  <a
    class="external-link"
    href="https://www.mousephenotype.org/data/genes/MGI:1860086"
    rel="noopener"
    target="_blank"
  >
    MGI:1860086
    <test-file-stub
      data-testid="external-link-icon"
      width="12.5"
    />
  </a>
</DocumentFragment>
`;

exports[`PublicationSource Exceptions renders correctly for MGI 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--computational-annotation"
    title="Computational annotation"
  >
    <test-file-stub />
  </span>
  MGI: 
  <a
    class="external-link"
    href="https://www.informatics.jax.org/marker/MGI:1860086"
    rel="noopener"
    target="_blank"
  >
    1860086
    <test-file-stub
      data-testid="external-link-icon"
      width="12.5"
    />
  </a>
</DocumentFragment>
`;

exports[`PublicationSource Normal xref renders but without links for unknown DBs 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--computational-annotation"
    title="Computational annotation"
  >
    <test-file-stub />
  </span>
  not-a-db: ID1234
</DocumentFragment>
`;

exports[`PublicationSource Normal xref renders correctly for normal xrefs 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--computational-annotation"
    title="Computational annotation"
  >
    <test-file-stub />
  </span>
  IntAct: 
  <a
    class="external-link"
    href="https://www.ebi.ac.uk/intact/search?query=id:P12345*#interactor"
    rel="noopener"
    target="_blank"
  >
    P12345
    <test-file-stub
      data-testid="external-link-icon"
      width="12.5"
    />
  </a>
</DocumentFragment>
`;

exports[`PublicationSource UniProtKB renders UniProtKB reviewed 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--reviewed"
    title="This marks a reviewed UniProtKB entry"
  >
    <test-file-stub />
  </span>
  UniProtKB reviewed (Swiss-Prot)
</DocumentFragment>
`;

exports[`PublicationSource UniProtKB renders UniProtKB unreviewed 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--unreviewed"
    title="This marks an unreviewed UniProtKB entry"
  >
    <test-file-stub />
  </span>
  UniProtKB unreviewed (TrEMBL)
</DocumentFragment>
`;
